resource/errors.go

```diff
@@ -68,6 +68,13 @@ func (e *mustRebuildError) Error() string {
 	return fmt.Sprintf("cannot reconfigure %q; must rebuild", e.name)
 }
 
+// NewBuildTimeoutError is used when a resource times out during construction or reconfiguration.
+func NewBuildTimeoutError(name Name) error {
+	envVar := "VIAM_RESOURCE_CONFIGURATION_TIMEOUT"
```

Comment:
should we pull this out into a constant somewhere?

Reply:
We definitely could! I decided against it for a couple reasons.

1. It only pops up a couple places, and I don't expect it to continue to do so
2. the natural package to put it in (imo) was `robotimpl` but then it should probably be private, so inaccessible to `resource`. the second natural location (imo) was `robot`, but this creates a dependency loop with `resource`. putting it in `resource` seemed to be awkward, and putting it anywhere else didn't really make sense because it isn't used anywhere else.
3. It seems common in RDK to hard code certain strings, so I wasn't too concerned about this coming off as a code smell

None of these are super strong opinions, however. Definitely happy to revisit.

Reply:
sg